crates/cli/src/subcommands/publish.rs

```diff
@@ -137,10 +145,16 @@ pub async fn exec(mut config: Config, args: &ArgMatches) -> Result<(), anyhow::E
         query_params.push(("trace_log", "true"));
     }
 
-    let path_to_wasm = if !path_to_project.is_dir() && path_to_project.extension().map_or(false, |ext| ext == "wasm") {
-        path_to_project.clone()
+    let path_to_wasm;
+    if !path_to_project.is_dir() && path_to_project.extension().map_or(false, |ext| ext == "wasm") {
+        println!("Note: Using --project-path to provide a wasm file is deprecated, and will be");
+        println!("removed in a future release. Please use --wasm-file instead.");
```

Comment:
One last request, could we make a new PR which is breaking which removes this (mark the PR with release 0.9 and ABI/API break)

Reply:
WIP but PR started here https://github.com/clockworklabs/SpacetimeDB/pull/915